### ISSUE TITLE
in upgrade-14, Zoe doesn't record new bundleCap when contracts are upgraded

### ISSUE THREAD

**Issue by Chris-Hibbert** [Agoric/Agoric#9249]
## Describe the bug

When a contract is upgraded, Zoe should remember the bundleCap, so that `E(adminFacet).restartContract(privateArgs);` will get the new bundle.

This was identified while  testing upgrades in #8911, and a fix is included there that has been merged to master. This fix was not included in upgrade 14, so contracts have to be upgraded using `E(adminFacet).upgradeContract()`, which requires specifying the bundle.

## To Reproduce

#9204 was blocked on this. [probeZcfBundle.js](https://github.com/Agoric/agoric-sdk/blob/e2e36cc1e1c015fc517a179019184fdc81f3e809/packages/vats/src/proposals/probeZcfBundle.js#L51) upgrades walletFactory twice, and uses `restartContract` the second time

## Expected behavior

If Zoe does `state.contractBundleCap = newContractBundleCap;` as in #8911, `restartContract()`  will re-use the most recent bundle.

**Comment by Chris-Hibbert** [comment]
As long as this hasn't been merged into MainNet, we have to remember that contract upgrades don't record the new bundleCap, so `restartContract()` will revert to an earlier version. The earlier we can get the fix merged, the better.

**Comment by mhofman** [comment]
We should likely include this in a chain software upgrade sooner rather than later since any contracts upgraded before this is deployed will not have their bundleID remembered, and will try to revert the originally installed bundle when doing a `restartContract`. Maybe in upgrade-16?

We do need to run the repair KREAd step along side this as it requires a Zoe upgrade.

**Comment by mhofman** [comment]
> As long as this hasn't been merged to master, we have to remember that contract upgrades don't record the new bundleCap, so `restartContract()` will revert to an earlier version. The earlier we can get the fix merged, the better.

s/merged to master/included in a release/ ?

**Comment by Chris-Hibbert** [comment]
> s/merged to master/included in a release/ ?
corrected.  thanks.

**Comment by Chris-Hibbert** [comment]
[Fixed (as a drive-by) i](https://github.com/Agoric/agoric-sdk/blame/7296b4772c47c36a28c28e8e8f13c74c7b3fc697/packages/zoe/src/zoeService/startInstance.js#L267)n #8911, and pushed to master with upgrade 16.
